### PR TITLE
awsbck: init at 0.3.2

### PR DESCRIPTION
**Pull request by beeb**
###### Description of changes

Adds `awsbck`, a utility to backup a local folder to an AWS S3 bucket periodically.

https://github.com/beeb/awsbck

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/backup/awsbck/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "awsbck";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "beeb";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-4iFPHMCWKOfwqdjCLQqWHSs5SwXi+K2sQu75ecsolSs=";
```
Comment (drupol) [inline pkgs/tools/backup/awsbck/default.nix:17]:
```suggestion
    hash = "sha256-4iFPHMCWKOfwqdjCLQqWHSs5SwXi+K2sQu75ecsolSs=";
```

**Review by drupol — CHANGES_REQUESTED** [review]
A minor change and it's good to go.

File: pkgs/tools/backup/awsbck/default.nix (hunk 1 above)
Comment (beeb) [reply pkgs/tools/backup/awsbck/default.nix]: Done! Sorry for the rebases and force pushes due to formatting shenanigans, finally got to the state I wanted.

File: pkgs/tools/backup/awsbck/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+, Security
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "awsbck";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "beeb";
+    repo = pname;
```
Comment (drupol) [inline pkgs/tools/backup/awsbck/default.nix:15]: Don't use `pname` inside `fetchFromGitHub`.

```suggestion
    repo = "awsbck";
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/tools/backup/awsbck/default.nix (hunk 2 above)
Comment (beeb) [reply pkgs/tools/backup/awsbck/default.nix]: Not sure why as they are equivalent, but done anyway! Explanation welcome :)

Comment (drupol) [reply pkgs/tools/backup/awsbck/default.nix]: I don't know the exact reason (I actually forgo). Apparently this is a trend in `nixpkgs`, if you look at already merged PRs, you'll see that other maintainers are asking contributors to do so.

**Pull request merged**